contract/src/lib.rs

```diff
@@ -337,6 +340,10 @@ impl MpcContract {
                     PromiseOrValue::Value(signature)
                 }
                 None => {
+                    if depth > 30 {
```

Comment:
why specifically 30? The max theoretical depth is ~60 and the max seen has been around late 50s. So why not 50?

Reply:
I've checked several transactions, and most of them ranged from 35 to 45. So I picked a reliable number. We do not want to have "out of gas" errors anymore. In good conditions we are returning a signature on step 8.